tempesta_fw/http.c

```diff
@@ -3172,7 +3192,13 @@ tfw_http_resp_process(TfwConn *conn, const TfwFsmData *data)
 
 	return r;
 bad_msg:
+	/*
+	 * Send error response for the bad requests if necessary.
+	 * In any case remove the request form forward and nip queues -
+	 * we won't resend it.
+	 */
 	bad_req = hmresp->req;
+	tfw_http_req_delist((TfwSrvConn *)conn, bad_req);
```

Comment:
My understanding is that we get here in all of the outcomes from the parser or GFSM: `PASS`, `BLOCK`, `POSTPONE`. My concern is a tricky case when the server considers a response NOT transmitted successfully. That may happen in `BLOCK` or `POSTPONE` cases. Will the server attempt to retransmit the response after the connection is restored? If so, the pairing of requests and responses must be maintained, and I am not sure that this situation is handled well in either patched or unpatched code.

Reply:
Good point, thanks @keshonok ! Issues #962 and #963 were created.